app/actions/action_deletePin.js

```diff
@@ -0,0 +1,18 @@
+import { DELETE_PIN } from '../constants/constants.js';
+import { userData } from '../lib/db/db.js';
+
+function deletePin (selectedPin) {
+  return {
+    type: DELETE_PIN,
+    payload: selectedPin
+  };
+};
+
+export default function (pin) {
+  return (dispatch) => {
+    userData.child(pin.id).remove();
+    dispatch(deletePin(pin));
+  };
+}
+
```

Comment:
If you want, I think this could be:
```JS
export default (pin) => (dispatch) => {
  userData.child(pin.id).remove();
  dispatch(deletePin(pin));
}
```